### ISSUE TITLE
open config yangs are not parsing with yang-js

### ISSUE THREAD

**Issue by sirikothe** [corenova/yang-js#49]
I am trying to parse open-config yangs using this library. But it is throwing some errors. These yangs compile fine with other compilers.
Some examples of the errors:
1. Following yang: 
https://github.com/openconfig/public/blob/master/release/models/interfaces/openconfig-interfaces.yang

throws following error:
ExpressionError: [module(openconfig-interfaces)/grouping(interface-ref-common)/leaf(subinterface)/type(leafref)/path] invalid path expression 'oc-if:interface[oc-if:name=current()' found in /oc-if:interfaces/oc-if:interface[oc-if:name=current()/../interface]/oc-if:subinterfaces/oc-if:subinterface/oc-if:index

2. This yang: 
https://github.com/openconfig/public/blob/master/release/models/policy/openconfig-routing-policy.yang

throws following error:

ExpressionError: [module(openconfig-routing-policy)/grouping(generic-actions)/choice(route-disposition)] cannot contain more than one non-case data node statement

Please let me know I am doing something wrong.

**Comment by saintkepha** [comment]
@sirikothe - thanks for reporting the issues.  

The issue #1 is due to the fact that `yang-js` does not contain a complete implementation of XPATH processing (particularly dealing with predicates - the stuff between [ and ]).  I think it should generate a warning in this case instead of a fatal error. 

As to issue #2, I think the issue is with the openconfig YANG schema.  

If you refer to RFC 6020 section 7.9.2 (I should probably reference that in the error message) it states that we can treat a *single* data node element inside *choice* statement as an implicit *case*.  The openconfig YANG schema has **two** leaf statements directly inside a *choice* statement.

**Comment by sirikothe** [comment]
@saintkepha 
Thank you for your quick reply.
Regarding issue: 1, generating a warning is fine instead of fatal error so that users can continue.
Regarding issue 2, I think you are right. the spec is a bit ambiguous in this case. Because the table in section 7.9.1 (The choice's Substatements)) indicates that leaf's cardinality is 0..n. Where as section 7.9.2 indicate that there can be at most one leaf directly under choice.

**Comment by vcshox** [comment]
@saintkepha

Is there any XPath function (Ex: current(), re-match(), ... ) has been implemented?

**Comment by saintkepha** [comment]
No, the predicate support is very limited and not much in the way of XPATH functions support.  The key reason for this is because `yang-js` uses JS native structures internally and does not use XML DOM.  If you can help prioritize the most common/useful XPATH functions used in YANG schemas, that would be very helpful for planning future implementation roadmap.

**Comment by saintkepha** [comment]
Keeping it open to address the XPATH parsing error...
